opengovsg/plumber
# PLU-125: [EXCEL-10] Data classification checks

## Problem
We should reject processing files that are too sensitive.

## Solution
GovTech implements data classification through [sensitivity labels](https://learn.microsoft.com/en-us/purview/sensitivity-labels-sharepoint-onedrive-files). Thus, we'll build an allowlist of appropriate labels' GUIDs, and check each files' sensitivity label against this list.

Note that I store sensitivity label GUIDs in an env var instead of hard-coding it. This is because each M365 tenant will have different GUIDs, and I wanted to enable us to easily configure GUIDs for our local dev tenants.

**Rejecting too sensitive file**
<img width="500" src="https://github.com/opengovsg/plumber/assets/135598754/8a3f5efe-9c9c-442f-9b05-4c8ae14631d6" />

**Allowing appropriate sensitivity file**
<img width="500" src="https://github.com/opengovsg/plumber/assets/135598754/2a432379-c6b5-4f42-a830-a9fb44044bd5" />

## Tests
- Check that I can access files with appropriate sensitivity labels
- Check that working with files with labels not in the allowlist result in an error
- Check that files with no sensitivity labels result in an error

## Deploy Notes
**New environment variables**:
- `M365_${tenantKey}_ALLOWED_SENSITIVITY_LABEL_GUIDS_CSV`: A comma separated list of allowed sensitivity label GUIDs for a particular M365 tenant.
